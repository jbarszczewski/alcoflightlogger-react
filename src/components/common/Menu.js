import React from 'react';
import { Link } from 'react-router-dom';
import { Nav, Navbar, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';

const Menu = () => {
    return (
        <div>
            <Navbar inverse collapseOnSelect>
                <Navbar.Header>
                    <Navbar.Brand>
                        <Link to="/" activeclassname="active">AFL</Link>
                    </Navbar.Brand>
                    <Navbar.Toggle />
                </Navbar.Header>
                <Navbar.Collapse>
                    <Nav>
                        <LinkContainer to="/flights">
                            <NavItem eventKey={1}>Flights</NavItem>
                        </LinkContainer>
                        <LinkContainer to="/login">
                            <NavItem eventKey={2}>Login</NavItem>
                        </LinkContainer>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        </div>
    );
};

export default Menu;
