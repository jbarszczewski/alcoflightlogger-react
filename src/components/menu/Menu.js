import React from 'react';
import { Link } from 'react-router-dom';
import { Nav, Navbar, NavItem } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';

const Menu = () => {
    return (
        <div>
            <Navbar inverse collapseOnSelect>
                <Navbar.Header>
                    <Navbar.Brand>
                        <Link id="logo" to="/" activeclassname="active">afl</Link>
                    </Navbar.Brand>
                    <Navbar.Toggle />
                </Navbar.Header>
                <Navbar.Collapse>
                    <Nav>
                        <LinkContainer to="/flights">
                            <NavItem id="flights" eventKey={1}>flights</NavItem>
                        </LinkContainer>
                        <LinkContainer to="/login">
                            <NavItem id="login" eventKey={2}>log in</NavItem>
                        </LinkContainer>
                    </Nav>
                </Navbar.Collapse>
            </Navbar>
        </div>
    );
};

export default Menu;
