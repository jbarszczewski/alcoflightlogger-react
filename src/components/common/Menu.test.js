import React from 'react';
import { shallow } from 'enzyme';
import Menu from './Menu';

let wrapper;

describe('Menu component', () => {
    beforeEach(() => {
        wrapper = shallow(<Menu />);
    });

    it('renders without crashing', () => {
        expect(wrapper).toHaveLength(1);
    });

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});