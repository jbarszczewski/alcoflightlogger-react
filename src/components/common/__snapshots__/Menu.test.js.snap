// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Uncontrolled(Navbar)
        bsStyle="default"
        collapseOnSelect={true}
        inverse={true}
>
        <NavbarHeader>
                <NavbarBrand>
                        <Link
                                activeclassname="active"
                                replace={false}
                                to="/"
                        >
                                AFL
                        </Link>
                </NavbarBrand>
                <NavbarToggle />
        </NavbarHeader>
        <NavbarCollapse>
                <Nav
                        bsClass="nav"
                        justified={false}
                        pullLeft={false}
                        pullRight={false}
                        stacked={false}
                >
                        <LinkContainer
                                activeClassName="active"
                                exact={false}
                                replace={false}
                                strict={false}
                                to="/flights"
                        >
                                <NavItem
                                        active={false}
                                        disabled={false}
                                        eventKey={1}
                                >
                                        Flights
                                </NavItem>
                        </LinkContainer>
                        <LinkContainer
                                activeClassName="active"
                                exact={false}
                                replace={false}
                                strict={false}
                                to="/login"
                        >
                                <NavItem
                                        active={false}
                                        disabled={false}
                                        eventKey={2}
                                >
                                        Login
                                </NavItem>
                        </LinkContainer>
                </Nav>
        </NavbarCollapse>
</Uncontrolled(Navbar)>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsStyle": "default",
        "children": Array [
          <NavbarHeader>
            <NavbarBrand>
                        <Link
                                    activeclassname="active"
                                    replace={false}
                                    to="/"
                        >
                                    AFL
                        </Link>
            </NavbarBrand>
            <NavbarToggle />
</NavbarHeader>,
          <NavbarCollapse>
            <Nav
                        bsClass="nav"
                        justified={false}
                        pullLeft={false}
                        pullRight={false}
                        stacked={false}
            >
                        <LinkContainer
                                    activeClassName="active"
                                    exact={false}
                                    replace={false}
                                    strict={false}
                                    to="/flights"
                        >
                                    <NavItem
                                                active={false}
                                                disabled={false}
                                                eventKey={1}
                                    >
                                                Flights
                                    </NavItem>
                        </LinkContainer>
                        <LinkContainer
                                    activeClassName="active"
                                    exact={false}
                                    replace={false}
                                    strict={false}
                                    to="/login"
                        >
                                    <NavItem
                                                active={false}
                                                disabled={false}
                                                eventKey={2}
                                    >
                                                Login
                                    </NavItem>
                        </LinkContainer>
            </Nav>
</NavbarCollapse>,
        ],
        "collapseOnSelect": true,
        "inverse": true,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <NavbarBrand>
                <Link
                                activeclassname="active"
                                replace={false}
                                to="/"
                >
                                AFL
                </Link>
</NavbarBrand>,
              <NavbarToggle />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Link
                  activeclassname="active"
                  replace={false}
                  to="/"
>
                  AFL
</Link>,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeclassname": "active",
                  "children": "AFL",
                  "replace": false,
                  "to": "/",
                },
                "ref": null,
                "rendered": "AFL",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Nav
              bsClass="nav"
              justified={false}
              pullLeft={false}
              pullRight={false}
              stacked={false}
>
              <LinkContainer
                            activeClassName="active"
                            exact={false}
                            replace={false}
                            strict={false}
                            to="/flights"
              >
                            <NavItem
                                          active={false}
                                          disabled={false}
                                          eventKey={1}
                            >
                                          Flights
                            </NavItem>
              </LinkContainer>
              <LinkContainer
                            activeClassName="active"
                            exact={false}
                            replace={false}
                            strict={false}
                            to="/login"
              >
                            <NavItem
                                          active={false}
                                          disabled={false}
                                          eventKey={2}
                            >
                                          Login
                            </NavItem>
              </LinkContainer>
</Nav>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "nav",
              "children": Array [
                <LinkContainer
                  activeClassName="active"
                  exact={false}
                  replace={false}
                  strict={false}
                  to="/flights"
>
                  <NavItem
                                    active={false}
                                    disabled={false}
                                    eventKey={1}
                  >
                                    Flights
                  </NavItem>
</LinkContainer>,
                <LinkContainer
                  activeClassName="active"
                  exact={false}
                  replace={false}
                  strict={false}
                  to="/login"
>
                  <NavItem
                                    active={false}
                                    disabled={false}
                                    eventKey={2}
                  >
                                    Login
                  </NavItem>
</LinkContainer>,
              ],
              "justified": false,
              "pullLeft": false,
              "pullRight": false,
              "stacked": false,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeClassName": "active",
                  "children": <NavItem
                    active={false}
                    disabled={false}
                    eventKey={1}
>
                    Flights
</NavItem>,
                  "exact": false,
                  "replace": false,
                  "strict": false,
                  "to": "/flights",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "children": "Flights",
                    "disabled": false,
                    "eventKey": 1,
                  },
                  "ref": null,
                  "rendered": "Flights",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "activeClassName": "active",
                  "children": <NavItem
                    active={false}
                    disabled={false}
                    eventKey={2}
>
                    Login
</NavItem>,
                  "exact": false,
                  "replace": false,
                  "strict": false,
                  "to": "/login",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "children": "Login",
                    "disabled": false,
                    "eventKey": 2,
                  },
                  "ref": null,
                  "rendered": "Login",
                  "type": [Function],
                },
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Uncontrolled(Navbar)
          bsStyle="default"
          collapseOnSelect={true}
          inverse={true}
>
          <NavbarHeader>
                    <NavbarBrand>
                              <Link
                                        activeclassname="active"
                                        replace={false}
                                        to="/"
                              >
                                        AFL
                              </Link>
                    </NavbarBrand>
                    <NavbarToggle />
          </NavbarHeader>
          <NavbarCollapse>
                    <Nav
                              bsClass="nav"
                              justified={false}
                              pullLeft={false}
                              pullRight={false}
                              stacked={false}
                    >
                              <LinkContainer
                                        activeClassName="active"
                                        exact={false}
                                        replace={false}
                                        strict={false}
                                        to="/flights"
                              >
                                        <NavItem
                                                  active={false}
                                                  disabled={false}
                                                  eventKey={1}
                                        >
                                                  Flights
                                        </NavItem>
                              </LinkContainer>
                              <LinkContainer
                                        activeClassName="active"
                                        exact={false}
                                        replace={false}
                                        strict={false}
                                        to="/login"
                              >
                                        <NavItem
                                                  active={false}
                                                  disabled={false}
                                                  eventKey={2}
                                        >
                                                  Login
                                        </NavItem>
                              </LinkContainer>
                    </Nav>
          </NavbarCollapse>
</Uncontrolled(Navbar)>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsStyle": "default",
          "children": Array [
            <NavbarHeader>
              <NavbarBrand>
                            <Link
                                          activeclassname="active"
                                          replace={false}
                                          to="/"
                            >
                                          AFL
                            </Link>
              </NavbarBrand>
              <NavbarToggle />
</NavbarHeader>,
            <NavbarCollapse>
              <Nav
                            bsClass="nav"
                            justified={false}
                            pullLeft={false}
                            pullRight={false}
                            stacked={false}
              >
                            <LinkContainer
                                          activeClassName="active"
                                          exact={false}
                                          replace={false}
                                          strict={false}
                                          to="/flights"
                            >
                                          <NavItem
                                                        active={false}
                                                        disabled={false}
                                                        eventKey={1}
                                          >
                                                        Flights
                                          </NavItem>
                            </LinkContainer>
                            <LinkContainer
                                          activeClassName="active"
                                          exact={false}
                                          replace={false}
                                          strict={false}
                                          to="/login"
                            >
                                          <NavItem
                                                        active={false}
                                                        disabled={false}
                                                        eventKey={2}
                                          >
                                                        Login
                                          </NavItem>
                            </LinkContainer>
              </Nav>
</NavbarCollapse>,
          ],
          "collapseOnSelect": true,
          "inverse": true,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <NavbarBrand>
                  <Link
                                    activeclassname="active"
                                    replace={false}
                                    to="/"
                  >
                                    AFL
                  </Link>
</NavbarBrand>,
                <NavbarToggle />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": <Link
                    activeclassname="active"
                    replace={false}
                    to="/"
>
                    AFL
</Link>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeclassname": "active",
                    "children": "AFL",
                    "replace": false,
                    "to": "/",
                  },
                  "ref": null,
                  "rendered": "AFL",
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Nav
                bsClass="nav"
                justified={false}
                pullLeft={false}
                pullRight={false}
                stacked={false}
>
                <LinkContainer
                                activeClassName="active"
                                exact={false}
                                replace={false}
                                strict={false}
                                to="/flights"
                >
                                <NavItem
                                                active={false}
                                                disabled={false}
                                                eventKey={1}
                                >
                                                Flights
                                </NavItem>
                </LinkContainer>
                <LinkContainer
                                activeClassName="active"
                                exact={false}
                                replace={false}
                                strict={false}
                                to="/login"
                >
                                <NavItem
                                                active={false}
                                                disabled={false}
                                                eventKey={2}
                                >
                                                Login
                                </NavItem>
                </LinkContainer>
</Nav>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "nav",
                "children": Array [
                  <LinkContainer
                    activeClassName="active"
                    exact={false}
                    replace={false}
                    strict={false}
                    to="/flights"
>
                    <NavItem
                                        active={false}
                                        disabled={false}
                                        eventKey={1}
                    >
                                        Flights
                    </NavItem>
</LinkContainer>,
                  <LinkContainer
                    activeClassName="active"
                    exact={false}
                    replace={false}
                    strict={false}
                    to="/login"
>
                    <NavItem
                                        active={false}
                                        disabled={false}
                                        eventKey={2}
                    >
                                        Login
                    </NavItem>
</LinkContainer>,
                ],
                "justified": false,
                "pullLeft": false,
                "pullRight": false,
                "stacked": false,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeClassName": "active",
                    "children": <NavItem
                      active={false}
                      disabled={false}
                      eventKey={1}
>
                      Flights
</NavItem>,
                    "exact": false,
                    "replace": false,
                    "strict": false,
                    "to": "/flights",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "children": "Flights",
                      "disabled": false,
                      "eventKey": 1,
                    },
                    "ref": null,
                    "rendered": "Flights",
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "activeClassName": "active",
                    "children": <NavItem
                      active={false}
                      disabled={false}
                      eventKey={2}
>
                      Login
</NavItem>,
                    "exact": false,
                    "replace": false,
                    "strict": false,
                    "to": "/login",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "children": "Login",
                      "disabled": false,
                      "eventKey": 2,
                    },
                    "ref": null,
                    "rendered": "Login",
                    "type": [Function],
                  },
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
